src/commons/navigationBar/NavigationBar.tsx

```diff
@@ -65,277 +55,222 @@ const NavigationBar: React.FC = () => {
 
-  const enableDesktopPopoverIn = [
-    '/playground',
-    '/sicpjs',
-    '/contributors',
-    `/courses/${courseId}/sourcecast`,
-    `/courses/${courseId}/achievements`
-  ];
-  const enableDesktopPopover =
-    courseId != null && !!enableDesktopPopoverIn.find(x => location.pathname.startsWith(x));
+  const fullAcademyNavbarLeftInfoWithAssessments: NavbarEntryInfo[] = useMemo(() => {
+    return [
+      ...fullAcademyNavbarLeftAssessmentsInfo,
+      ...fullAcademyNavbarLeftCommonInfo,
+      ...fullAcademyNavbarLeftStaffInfo
+    ];
+  }, [
+    fullAcademyNavbarLeftAssessmentsInfo,
+    fullAcademyNavbarLeftCommonInfo,
+    fullAcademyNavbarLeftStaffInfo
+  ]);
 
-  // Handles the Source Academy @ NUS left desktop navbar group
-  const desktopNavbarLeft = (
+  const renderPlaygroundOnlyNavbarLeftDesktop = () => (
     <NavbarGroup align={Alignment.LEFT}>
-      {enableDesktopPopover ? (
+      {renderNavlinksFromInfo(playgroundOnlyNavbarLeftInfo, createDesktopNavlink)}
+    </NavbarGroup>
+  );
+
+  const renderPlaygroundOnlyNavbarLeftMobile = () =>
+    wrapWithMobileHamburger(
+      renderNavlinksFromInfo(playgroundOnlyNavbarLeftInfo, createMobileNavlink)
+    );
+
+  const renderFullAcademyNavbarLeftDesktop = () => {
+    const desktopNavbarLeftPopoverContent = (
+      <Navbar>
+        <NavbarGroup>
+          {renderNavlinksFromInfo(
+            assessmentTypesToNavlinkInfo({
+              assessmentTypes,
+              courseId,
+              isEnrolledInACourse
+            }),
+            createDesktopNavlink
+          )}
+        </NavbarGroup>
+      </Navbar>
+    );
+    const topNavbarNavlinks = [
+      '/playground',
+      '/sicpjs',
+      '/contributors',
+      '/githubassessments',
+      `/courses/${courseId}/sourcecast`,
+      `/courses/${courseId}/achievements`
+    ];
+    const enableDesktopPopover =
+      courseId != null && !!topNavbarNavlinks.find(x => location.pathname.startsWith(x));
+    const highlightDesktopLogo = (location: Location) => {
+      // Highlight main logo when none of the topmost-blue navbar links are active
+      return !topNavbarNavlinks.find(x => location.pathname.startsWith(x));
+    };
+
+    return (
+      <NavbarGroup align={Alignment.LEFT}>
         <Popover2
           position={Position.BOTTOM_RIGHT}
           interactionKind="hover"
           content={desktopNavbarLeftPopoverContent}
           popoverClassName={'desktop-navbar-popover'}
+          disabled={!enableDesktopPopover}
         >
-          {desktopLogoButton}
+          <NavLink
+            className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL, {
+              [Classes.ACTIVE]: highlightDesktopLogo(location)
+            })}
+            to={courseId == null ? '/welcome' : `/courses/${courseId}`}
+          >
+            <Icon icon={IconNames.SYMBOL_DIAMOND} />
+            <NavbarHeading style={{ paddingBottom: '0px' }}>
+              {courseShortName || 'Source Academy'}
```

Comment:
Ditto. Noticed that "Source Academy @ NUS" is also no more. Perhaps the default fallback value can be done conditionally?

Reply:
This is done on purpose to take out the hardcoded @ NUS. Replaced with a more general purposed 'Source Academy'

Reply:
Noted, that makes sense. I'm just thinking of potential integrations with i18n, hence my suggestion on putting this into `Constants.tsx` or something like that. But on second though, let us not prematurely do all these refactoring. Thanks!